src/arrayMethodJoin.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+    const stringSep = typeof (separator) === typeof (undefined)
+      ? ','
+      : String(separator);
+
+    for (let x = 0; x < this.length; x++) {
+      if (this[x] === undefined || this[x] === null) {
+        this[x] = '';
+      }
+
+      res += this[x]
+      + (x === this.length - 1
+        ? ''
+        : stringSep);
```

Comment:
instead of adding this weird ternary operator, better invert your first if statement and add element is it's NOT null or undefined, and then only add separator if it's not last item